From 1974ab1c183f04b180bd1c6d4ea3339578bd71f1 Mon Sep 17 00:00:00 2001
From: Sudhanshu Gupta <sudhanshu.gupta@nxp.com>
Date: Wed, 26 Jul 2017 16:48:31 +0530
Subject: [PATCH 025/706] arm64: dts: ls1028a: correct PCI node properties in
 device tree

Signed-off-by: Sudhanshu Gupta <sudhanshu.gupta@nxp.com>
(cherry picked from commit db99f4cc8a9443256295eab9abc9bb3973bc0721)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 28 +++++++++++--------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 0433ebecfd6b..3e3e5c4fe061 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -277,11 +277,13 @@
 	};
 
 	pcie@3400000 {
-		compatible = "fsl,ls2085a-pcie", "fsl,ls1088a-pcie",
+		compatible = "fsl,ls-pcie", "fsl,ls2085a-pcie", "fsl,ls1088a-pcie",
 			     "snps,dw-pcie";
-		reg = <0x00 0x03400000 0x0 0x00100000 /* controller registers */
-		       0x20 0x00000000 0x0 0x00002000>;/* configuration space */
-		reg-names = "regs", "config";
+		reg = <0x00 0x03400000 0x0 0x80000
+			0x00 0x03480000 0x0 0x40000   /* lut registers */
+			0x00 0x034c0000 0x0 0x40000  /* pf controls registers */
+			0x80 0x00000000 0x0 0x20000>; /* configuration space */
+		reg-names = "dbi", "lut", "ctrl", "config";
 		interrupts = <0 108 0x4>; /* aer interrupt */
 		interrupt-names = "aer";
 		#address-cells = <3>;
@@ -290,8 +292,8 @@
 		dma-coherent;
 		num-lanes = <4>;
 		bus-range = <0x0 0xff>;
-		ranges = <0x81000000 0x0 0x00000000 0x20 0x00010000 0x0 0x00010000   /* downstream I/O */
-			  0x82000000 0x0 0x40000000 0x20 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
+		ranges = <0x81000000 0x0 0x00000000 0x80 0x00020000 0x0 0x00010000   /* downstream I/O */
+			  0x82000000 0x0 0x40000000 0x80 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
 		msi-parent = <&its>;
 		#interrupt-cells = <1>;
 		interrupt-map-mask = <0 0 0 7>;
@@ -302,11 +304,13 @@
 	};
 
 	pcie@3500000 {
-		compatible = "fsl,ls2085a-pcie", "fsl,ls1088a-pcie",
+		compatible = "fsl,ls-pcie", "fsl,ls2085a-pcie", "fsl,ls1088a-pcie",
 			     "snps,dw-pcie";
-		reg = <0x00 0x03500000 0x0 0x00100000 /* controller registers */
-		       0x28 0x00000000 0x0 0x00002000>;/* configuration space */
-		reg-names = "regs", "config";
+		reg = <0x00 0x03500000 0x0 0x80000
+			0x00 0x03580000 0x0 0x40000   /* lut registers */
+			0x00 0x035c0000 0x0 0x40000  /* pf controls registers */
+			0x88 0x00000000 0x0 0x20000>; /* configuration space */
+		reg-names = "dbi", "lut", "ctrl", "config";
 		interrupts = <0 113 0x4>; /* aer interrupt */
 		interrupt-names = "aer";
 		#address-cells = <3>;
@@ -315,8 +319,8 @@
 		dma-coherent;
 		num-lanes = <4>;
 		bus-range = <0x0 0xff>;
-		ranges = <0x81000000 0x0 0x00000000 0x28 0x00010000 0x0 0x00010000   /* downstream I/O */
-			  0x82000000 0x0 0x40000000 0x28 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
+		ranges = <0x81000000 0x0 0x00000000 0x88 0x00020000 0x0 0x00010000   /* downstream I/O */
+			  0x82000000 0x0 0x40000000 0x88 0x40000000 0x0 0x40000000>; /* non-prefetchable memory */
 		msi-parent = <&its>;
 		#interrupt-cells = <1>;
 		interrupt-map-mask = <0 0 0 7>;
-- 
2.17.1

